### PR TITLE
Should send all sched commands via secondary sock

### PR DESCRIPTION
**Pull request by agrawalravi90**
#### Describe Bug or Feature

This is hopefully the fix for stat issues. When we use the primary connection to communicate both the sched commands and the IFL replies, sometimes they can get mixed up. Here's an example:
server 1 triggers a sched cycle
server 2 also triggers a sched cycle, so the tcp buffers have a sched command to read for the scheduler
scheduler makes an IFL call to server 2 because of server 1's trigger
instead of a string reply, it gets an int sched command value from DIS, causing the issue

Kudos to Nithin for finding this issue.

#### Describe Your Change

Changing things so that ALL scheduler commands are now communicated via the secondary fd, leaving the primary fd for IFL. This will separate the IFL data from scheduler commands data.

### CODE REVIEW

File: src/scheduler/pbs_sched.c [hunk 1]
```diff
@@ -1478,6 +1478,15 @@ socket_to_conn(int sock, struct sockaddr_in saddr_in)
 	if ((conn_arr = get_conn_servers()) == NULL)
 		return -1;
 
+	conn_arr[svr_conn_index].port = port;
+
+	if (conn_arr[svr_conn_index].sd != -1 && conn_arr[svr_conn_index].secondary_sd != -1) {
+		/* Receiving new connection request from a server who was connected previously */
+		conn_arr[svr_conn_index].sd = -1;
+		conn_arr[svr_conn_index].secondary_sd = -1;
+		conn_arr[svr_conn_index].state = SVR_CONN_STATE_DOWN;
```
Comment (suresh-thelkar) [inline src/scheduler/pbs_sched.c:1487]: Just curious to know why this change is needed. If Server is connected(primary+secondary) to scheduler it does not again try to connect, unless sched closes its connections that case is already handled. Also directly setting sd and secondary_sd to -1 cause problems because Server does not get a notification that sched closed its connections.

**Comment by suresh-thelkar** [comment]
@subhasisb, As discussed I am taking the changes in this PR and addressing few more bugs. So please do not merge this PR. I will raise new PR on top of this PR changes and let you know.

File: src/scheduler/pbs_sched.c (hunk 1 above)
Comment (agrawalravi90) [reply src/scheduler/pbs_sched.c:1487]: I noticed that if we bring the server down and up again without restarting the scheduler, the server connections array didn't get updated automatically .. this means that when the server reached out to scheduler again for the first time for primary connection, scheduler will check if (sd == -1), which will not be -1, so it would assigned secondary_sd to the primary connection, which resulted in problems

Comment (agrawalravi90) [reply src/scheduler/pbs_sched.c:1487]: "unless sched closes its connections that case is already handled"

can you please verify this? I didn't see it get handled automatically

Comment (suresh-thelkar) [reply src/scheduler/pbs_sched.c:1487]: I found the root cause and fixing it. This is happening because sched connections are not getting closed properly as we are now sending sched commands on secondary connections. Please look into my upcoming PR for the changes.
